src/modules/muorb/krait/px4muorb_KraitRpcWrapper.cpp

```diff
@@ -122,13 +195,32 @@ bool px4muorb::KraitRpcWrapper::Initialize()
 		PX4_DEBUG("%s rpcmem_alloc passed for data_buffer", __FUNCTION__);
 	}
 
-	// call myorb intiialize rotine.
+	int32_t time_diff_us;
+
+	if (calc_timer_diff_to_dsp_us(&time_diff_us) != 0) {
+		rc = false;
+		return rc;
+	}
+
+	// call muorb initialize routine.
 	if (px4muorb_orb_initialize() != 0) {
 		PX4_ERR("%s Error calling the uorb fastrpc initalize method..", __FUNCTION__);
 		rc = false;
 		return rc;
 	}
 
+	// TODO FIXME: remove this check or make it less verbose later
+	px4muorb_set_absolute_time_offset(time_diff_us);
+
+	uint64_t time_dsp;
+	px4muorb_get_absolute_time(&time_dsp);
+
+	uint64_t time_appsproc = hrt_absolute_time();
+
+	int diff = (time_dsp - time_appsproc);
+
+	PX4_INFO("time_dsp: %llu us, time appsproc: %llu us, diff: %d us", time_dsp, time_appsproc, diff);
```

Comment:
It seems like this test mostly counts the time needed for the get-time calls. Flipping lines 216 and 218 also inverts the sign of `diff`. The average of the two gives a difference of about +9us.